libparsec/crates/platform_device_loader/tests/recovery.rs

```diff
@@ -1,3 +1,43 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
 
-// TODO !
+#![allow(clippy::unwrap_used)]
+
+use libparsec_platform_device_loader::{export_recovery_device, import_recovery_device};
+use libparsec_tests_lite::parsec_test;
+use libparsec_types::HumanHandle;
+use libparsec_types::ParsecOrganizationAddr;
+use libparsec_types::UserProfile;
+use libparsec_types::{DeviceLabel, LocalDevice};
+
+#[parsec_test]
+async fn test_ok_recovery() {
+    let url = ParsecOrganizationAddr::from_any(
+        // cspell:disable-next-line
+        "parsec3://test.invalid/Org?no_ssl=true&p=xCD7SjlysFv3d4mTkRu-ZddRjIZPGraSjUnoOHT9s8rmLA",
+    )
+    .unwrap();
+    let local_device = LocalDevice::generate_new_device(
+        url,
+        UserProfile::Admin,
+        HumanHandle::new("alice@dev1", "alice").unwrap(),
+        "alice label".parse().unwrap(),
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+    );
+    let recovery_device_label = DeviceLabel::default();
+    let (secret_passphrase, data, exported_recovery_device) =
+        export_recovery_device(&local_device, recovery_device_label.clone())
+            .await
+            .unwrap();
+    let imported_recovery_device = import_recovery_device(&data, secret_passphrase)
+        .await
+        .unwrap();
+
+    assert_eq!(exported_recovery_device, imported_recovery_device);
+    assert_eq!(imported_recovery_device.device_label, recovery_device_label);
+}
```

Comment:
Given you have `secret_passphrase` and `data` available here, it's convenient to add test the error case for import at the end of this test function:
```rust

    // It's convenient to also test the import bad cases here

    // Invalid data
    p_assert_matches!(
        import_recovery_device(
            &b"dummy", secret_passphrase.clone()
        )
            .await,
        Err(ImportRecoveryDeviceError::InvalidData)
    );

    // Invalid passphrase

    p_assert_matches!(
        import_recovery_device(
            &data, b"dummy".to_string().into()
        )
            .await,
        Err(ImportRecoveryDeviceError::InvalidPassphrase)
    );

    // Decryption failed

    let different_secret_passphrase = {
        let mut different_secret_passphrase = secret_passphrase.clone();
        different_secret_passphrase[0] += 1; // Change the first byte
        different_secret_passphrase
    };
    p_assert_matches!(
        import_recovery_device(
            &data, different_secret_passphrase
        )
            .await,
        Err(ImportRecoveryDeviceError::DecryptionFailed)
    );
```